clang_build/target.py

```diff
@@ -81,8 +81,8 @@ def __init__(self,
             # Public include directories are forwarded
             self.include_directories_public += target.include_directories_public
 
-        self.include_directories = list(set([dir.resolve() for dir in self.include_directories]))
-        self.headers = list(set(self.headers))
+        self.include_directories = list(dict.fromkeys(dir.resolve() for dir in self.include_directories))
```

Comment:
Since this requires Python >= 3.7, should we note this requirement somewhere? Especially `pip install`ing this package might work, but then throw errors for users with older Python3 versions.

Reply:
Yes, I will change this.

Reply:
I added the requirement